.github/workflows/pull-request-docker.yml

```diff
@@ -0,0 +1,181 @@
+
+          VERSION="$(cat version.txt)"
+          DOCKER_VERSION="${VERSION%-SNAPSHOT}"
+          echo "DOCKER_VERSION=${DOCKER_VERSION}" >> ${GITHUB_ENV}
+          echo "DOCKER_IMAGE=localhost:5000/nessie-testing" >> ${GITHUB_ENV}
+
+      - name: Gradle / prepare
+        uses: gradle/gradle-build-action@v2
+        with:
+          cache-read-only: true
+          # Just fetch cache, defer build to "later"
+          arguments: projects
+
+      - name: Docker images publishing
+        env:
+          ARTIFACTS: ../build-artifacts
+        run: |
+          rm -rf "${ARTIFACTS}"
+          mkdir -p "${ARTIFACTS}"
+
+          tools/dockerbuild/build-push-images.sh \
+            -a "${ARTIFACTS}" \
+            -g ":nessie-quarkus" \
+            -p "servers/quarkus-server" \
+            -n \
+            ${DOCKER_IMAGE}
+          rm -rf "${ARTIFACTS}"
+
+      - name: Cleanup buildx
+        run: |
+          docker buildx use default
+          docker buildx rm nessiebuild
+
+      - name: Docker images exist test
+        run: |
+          docker pull ${DOCKER_IMAGE}:latest
+          docker pull ${DOCKER_IMAGE}:latest-java
+          docker pull ${DOCKER_IMAGE}:latest-native
+          docker pull ${DOCKER_IMAGE}:${DOCKER_VERSION}
+          docker pull ${DOCKER_IMAGE}:${DOCKER_VERSION}-native
+          docker pull ${DOCKER_IMAGE}:${DOCKER_VERSION}-java
+          cat <<! >> $GITHUB_STEP_SUMMARY
+          ## Docker images
+          
+          \`\`\`
+          $(docker images)
+          \`\`\`
+          !
+
+      - name: Check if Docker native image works
+        run: |
+          docker run --detach --name nessie ${DOCKER_IMAGE}:latest-native
+          echo "Let native Nessie Docker image run for one minute (to make sure it starts up fine)..."
+          for i in {1..60}; do
+            STATUS="$(docker container inspect nessie | jq -r '.[0].State.Status')"
+            if [[ ${STATUS} != "running" ]] ; then
+              echo "Nessie native Docker image stopped on its own ... a bug?" > /dev/stderr
+              docker logs nessie
+              cat <<! >> $GITHUB_STEP_SUMMARY
+              ## Nessie native Docker image FAILED
+
+              \`\`\`
+              $(docker logs nessie)
+              \`\`\`
+          !
+              exit 1
+            fi
+            sleep 1
+          done
+          echo "## Nessie native Docker image smoke test: PASSED" >> $GITHUB_STEP_SUMMARY  
+          echo "Nessie native Docker image smoke test: PASSED"
+          docker stop nessie
+          docker rm nessie
+
+      - name: Check if Docker Java image works
+        run: |
+          docker run --detach --name nessie ${DOCKER_IMAGE}:latest-java
+          echo "Let Nessie Java Docker image run for one minute (to make sure it starts up fine)..."
+          for i in {1..60}; do
+            STATUS="$(docker container inspect nessie | jq -r '.[0].State.Status')"
```

Comment:
could there be value to run a HTTP GET with curl against our API endpoints as well?
maybe there could be reasons the container is still running but the application or http endpoints are dead

maybe our Dockerfile could contain a https://docs.docker.com/engine/reference/builder/#healthcheck for this purpose, but potentially as a followup

Reply:
Wanna take a stab on that?

Reply:
time probably doesnt allow it anytime soon... do we already have an endpoint that would be always available without auth?

Reply:
Heath checks probably.